### PR TITLE
Change mute, block and domain block confirmations in web UI

### PR DESCRIPTION
**Pull request by Gargron**
<video src="https://github.com/mastodon/mastodon/assets/184731/ad790281-ae71-450e-9baa-3de72f4a32d6" />

### CODE REVIEW

**Review by ClearlyClaire — COMMENTED** [review]
I like how it looks and behaves (although I find the transparency a little distracting, though not bad enough to hinder legibility), and the code is cleaner as well!

However, I have some concerns with the copy:
- ~~“Nobody from this server can mention or follow you.” when blocking a domain: the “Nobody from this server […] can mention you” part is a stronger assertion than what we can ensure~~ (EDIT: changed in latest push)
- “They can't see your posts and you won't see theirs.” when blocking a user: the “They can't see your posts” is a stronger assertion than what we can ensure (logged out view, or server software ignoring `Block` activities)
- “They can't mention or follow you.” when blocking a user: “They can't mention […] you” is a stronger assertion than what we can ensure across server boundaries

**Review by ClearlyClaire — APPROVED** [review]
Looks good overall, but I think the modal for blocks is over-promising things we cannot enforce. I'd be much more comfortable if it mentioned those caveats somehow, either directly or in a linked documentation page.

File: app/javascript/mastodon/features/ui/components/mute_modal.jsx [hunk 1]
```diff
@@ -1,138 +1,154 @@
...
+        </div>
+      </div>
+
+      <div className={classNames('safety-action-modal__bottom', { active: expanded })}>
+        <div className='safety-action-modal__bottom__collapsible'>
+          <div className='safety-action-modal__field-group'>
+            <RadioButtonLabel name='duration' value='0' label={intl.formatMessage(messages.indefinite)} currentValue={muteDuration} onChange={handleChangeMuteDuration} />
+            <RadioButtonLabel name='duration' value='86400' label={intl.formatMessage(messages.hours, { number: 24 })} currentValue={muteDuration} onChange={handleChangeMuteDuration} />
+            <RadioButtonLabel name='duration' value='604800' label={intl.formatMessage(messages.days, { number: 7 })} currentValue={muteDuration} onChange={handleChangeMuteDuration} />
+            <RadioButtonLabel name='duration' value='x' label={intl.formatMessage(messages.days, { number: 30 })} currentValue={muteDuration} onChange={handleChangeMuteDuration} />
```
Comment (tribela) [inline app/javascript/mastodon/features/ui/components/mute_modal.jsx:121]: `'x'`?

Comment (ClearlyClaire) [reply app/javascript/mastodon/features/ui/components/mute_modal.jsx:121]: Oops! Sorry I've missed that.

Comment (BenAveling) [reply app/javascript/mastodon/features/ui/components/mute_modal.jsx:121]: Is the plan for these values to be hard coded or will there be an option for instance admins to configure their own values?

Comment (ClearlyClaire) [reply app/javascript/mastodon/features/ui/components/mute_modal.jsx:121]: There is no plan to have those configurable at the moment.

**Comment by Joshix-1** [comment]
It's really annoying that previously available options get removed and we now have to click one extra time more to mute people for 24h.

I do not get why the options are hidden per default. There should be always enough space to display the whole dialog, so why introduce the extra click? For new users it may not even be obvious that there are options. I certainly wouldn't see and click the click options. I'd probably just complain that there is no option to configure the duration of mutes.

Once I've read what muting a user means, I do not have to read it again. I then know it. So why do I have to read it everytime? The thing I want to see everytime is the option to configure the mute duration. Because that's something I want to decide for every mute. If you want to keep the menu small, I'd always show the options and put the explanation behind a click. I can remember what muting means.

Also I've heard from at-least one person that they like to mute people for less than a day, so why remove that feature?

**Pull request merged**